### ISSUE TITLE
Convert database.xml to migrations

### ISSUE THREAD

**Issue by PVince81** [owncloud/music#865]
As of Nextcloud 22, the database.xml file has been deprecated:
```
% occ app:enable music
The appinfo/database.xml file is not longer supported. Used in music
```

we should migrate it to DB migrations

**Comment by PVince81** [comment]
see https://docs.nextcloud.com/server/21/developer_manual/basics/storage/migrations.html#migrate-from-database-xml

requires NC 21 as the command was removed in NC 22

**Comment by paulijar** [comment]
Actually, that has already been handled by the commit 6982d91b. But I guess I need to make a new Music app release including this commit soon, as the release of NC22 is closing in. Any idea about when the RC1 will be available? I wouldn't want to claim support for NC22 before actually testing with a release candidate; it has happened before that the compatibility has broken between the beta releases.

**Comment by PVince81** [comment]
Ah, nice! NC 22 is currently on 22.0.0beta5, not sure about the timeline for RC1 though.
Considering that many betas I guess it should be in the coming weeks.

**Comment by PVince81** [comment]
@paulijar RC1 is out: https://download.nextcloud.com/server/prereleases/nextcloud-22.0.0rc1.tar.bz2

**Comment by paulijar** [comment]
Okay, thanks for the info!

**Comment by paulijar** [comment]
Music v1.2.1 with support for NC22 is now released.

**Comment by PVince81** [comment]
Thanks for the release! :+1:

I've updated to that version from NC 21 and did a quick test with the following, which worked fine:
- search in web UI
- playback in web UI
- playlist listing and playback

also worked with the Ultrasonic Android app for playlist listing and playback.

**Comment by stefan123t** [comment]
@paulijar, thanks for your commit, this is very instructive for other apps too!

**Comment by paulijar** [comment]
@stefan123t No problem. If other apps use this as reference, then they should notice that there are some pit holes in the automatically generated Migration script as included in 6982d91. It does work correctly when installing on a clean environment, or an environment which already has an up-to-date database schema. But it cannot handle situations where there is an outdated DB schema from an older version of the app, and some existing table would need to be modified by adding and/or removing columns or indexes. 

To overcome this, my final implementation for the first Migrations script is like this: https://github.com/owncloud/music/blob/master/lib/Migration/Version010300Date20210906093000.php. It's fully hand-crafted although based on the automatically generated first version. It should now do the job, no matter which previous version of the Music app the user might have installed.
